cli/command/context/create.go

```diff
@@ -0,0 +1,222 @@
+package context
+
+import (
+	"io/ioutil"
+	"os"
+
+	"github.com/docker/cli/cli"
+	"github.com/docker/cli/cli/command"
+	"github.com/docker/cli/cli/context/docker"
+	"github.com/docker/cli/cli/context/kubernetes"
+	"github.com/docker/cli/cli/context/store"
+	"github.com/pkg/errors"
+	"github.com/spf13/cobra"
+	"github.com/spf13/pflag"
+)
+
+type createOptions struct {
+	name                     string
+	description              string // --description string: set the description
+	defaultStackOrchestrator string
+	docker                   dockerEndpointOptions
+	kubernetes               kubernetedEndpointOptions
+}
+
+func (o *createOptions) addFlags(flags *pflag.FlagSet) {
+	flags.StringVar(&o.description, "description", "", "set the description of the context")
+	flags.StringVar(&o.defaultStackOrchestrator, "default-stack-orchestrator", "", "set the default orchestrator for stack operations if different to the default one, to use with this context (swarm|kubernetes|all)")
+	o.docker.addFlags(flags)
+	o.kubernetes.addFlags(flags)
+}
+
+type dockerEndpointOptions struct {
+	host          string //--moby-host string: set moby endpoint host (same format as DOCKER_HOST)
+	apiVersion    string
+	ca            string //--moby-ca string: path to a CA file
+	cert          string //--moby-cert string: path to a client cert file
+	key           string //--moby-key string: path to a client key file
+	skipTLSVerify bool
+}
+
+func (o *dockerEndpointOptions) addFlags(flags *pflag.FlagSet) {
+	flags.StringVar(&o.host, "docker-host", "", "required: specify the docker endpoint on wich to connect")
```

Comment:
I don't like required flags, I think it should be an argument if it's mandatory. cc @thaJeztah @vdemeester

Reply:
I don't like commands with more than one unnamed argument (except when it is multiple arguments of the same nature like `docker image rm`)

Reply:
Actually the flag is not required. If not set, we use the default host value (just like when running `docker ...` without -H.